tools/pythonpkg/src/include/duckdb_python/filesystem_object.hpp

```diff
@@ -20,7 +20,8 @@ class FileSystemObject : public RegisteredObject {
 	virtual ~FileSystemObject() {
 		py::gil_scoped_acquire acquire;
 		// Assert that the 'obj' is a filesystem
-		D_ASSERT(py::isinstance<ModifiedMemoryFileSystem>(obj));
```

Comment:
This method should work if you implement check_ on the ModifiedMemoryFileSystem class?

Reply:
It has one
```c++
class ModifiedMemoryFileSystem : public py::object {
public:
	using py::object::object;
	ModifiedMemoryFileSystem(py::object object) : py::object(object) {
	}

public:
	static bool check_(const py::handle &object) {
		return py::isinstance(object, py::module::import("pyduckdb.filesystem").attr("ModifiedMemoryFileSystem"));
	}
};
```